### PR TITLE
Adds support for existing sirius-search indices

### PR DESCRIPTION
**Pull request by jmuscireum**

### CODE REVIEW

**Comment by jmuscireum** [comment]
BUILD

File: src/main/java/sirius/db/es/DefaultIndexNaming.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+/**
+ * Supplies default naming conventions for indices and mappings.
+ */
+@Register
+public class DefaultIndexNaming implements IndexNaming {
+
+    @Override
+    public int getPriority() {
+        return DEFAULT_PRIORITY;
```
Comment (sabieber) [inline src/main/java/sirius/db/es/DefaultIndexNaming.java]: I would maybe choose a lower priority

File: src/test/java/sirius/db/es/SuggestSpec.groovy [hunk 2]
```diff
@@ -55,13 +58,16 @@ class SuggestSpec extends BaseSpecification {
...
     def "phrase suggest works"() {
         when:
-        def entity1 = new SuggestTestEntity()
-        entity1.setContent("HSS drill bit")
-        elastic.update(entity1)
-        and:
-        def entity2 = new SuggestTestEntity()
-        entity2.setContent("Salmon with dill")
-        elastic.update(entity2)
+        // Add enough data for black magic
```
Comment (andyHa) [inline src/test/java/sirius/db/es/SuggestSpec.groovy]: is that supposed to be helpful?

File: src/main/java/sirius/db/mixing/properties/IntegerProperty.java [hunk 3]
```diff
@@ -79,6 +79,10 @@ protected Object transformFromDatasource(Class<? extends BaseMapper<?, ?, ?>> ma
             return Integer.valueOf(((Long) object.get()).intValue());
         }
 
+        if (object.get() instanceof String) {
```
Comment (andyHa) [inline src/main/java/sirius/db/mixing/properties/IntegerProperty.java]: if (object.is(String.class) { return object.asInt(); }

File: src/main/java/sirius/db/es/IndexNaming.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+ * http://www.scireum.de - info@scireum.de
+ */
+
+package sirius.db.es;
+
+import sirius.db.mixing.EntityDescriptor;
+
+/**
+ * Can be supplied to enforce naming conventions for indices and mappings in ElasticSearch on a per product basis.
+ */
```
Comment (andyHa) [inline src/main/java/sirius/db/es/IndexNaming.java]:
```suggestion
 * <p>
 * Use {@link sirius.kernel.di.std.Register} to make a custom implementation visible to the framework.
 */
```

**Pull request merged**